File: ui/grocery-list/src/app/services/grocery-item.service.ts
import { Injectable } from '@angular/core';
import { HttpHeaders, HttpClient } from '@angular/common/http';
import { apiConfig } from 'src/environments/environment';
import { GroceryItem } from '../models/grocery-item';
import { Observable } from 'rxjs';

const httpOptions = {
  headers: new HttpHeaders({ 'Content-Type': 'application/json' })
};

@Injectable({
  providedIn: "root"
})
export class GroceryItemService {
  apiUrl = `${apiConfig.url}items/`;

  constructor(private http: HttpClient) {}

  addItem(item: GroceryItem): Observable<GroceryItem> {
    return this.http.post<GroceryItem>(this.apiUrl, item, httpOptions);
  }

  removeItem(itemId: number): Observable<void> {
    return this.http.delete<void>(`${this.apiUrl}${itemId}`, httpOptions);
  }
}

File: Grocery-List/src/main/java/com/revature/grocery/list/service/GroceryItemService.java
package com.revature.grocery.list.service;

import com.revature.grocery.list.model.GroceryItem;

public interface GroceryItemService {

	public GroceryItem addGroceryItemToList(GroceryItem groceryItem);
	
	public boolean removeGroceryItemFromList(Integer itemId);
}
